generator/Can_Library_c.py

```diff
@@ -1,42 +1,46 @@
-"""
+'''
 Generate Can_Libary.c file.
 Run this file (with the spec path as a command line argument) to write just
 Can_Libary.c or main.py to write all files.
-"""
+'''
 import sys
-sys.path.append("ParseCAN")
+sys.path.append('ParseCAN')
 import ParseCAN
 from math import ceil, floor, log2
-from common import can_lib_c_path, can_lib_c_base_path
+from common import can_lib_c_path, can_lib_c_base_path, coord
 
 
-def write(output_path, spec_path, base_path):
-    """
+def write(car, output_path=can_lib_c_path, base_path=can_lib_c_base_path):
+    '''
     Generate Can_Libary.c file.
 
     :param output_path: file to be written to
-    :param spec_path: CAN spec path
+    :param car: CAN spec
     :param base_path: File with template code that's not autogenerated
-    """
-    car = ParseCAN.spec.car(spec_path)
+    '''
     with open(output_path, 'w') as f:
-        f.write('#include "Can_Library.h"\n')
+        fw = f.write
+
+        fw('#include "Can_Library.h"\n')
 
         # Copy over base
         with open(base_path) as base:
             lines = base.readlines()
             f.writelines(lines)
-        f.write("\n")
+
+        fw('\n')
 
         # Write init functions
         for board in car.boards:
             if board.arch:  # Means it's a board we program
                 for bus in board.subscribe:
-                    f.write("void " + bus.name.title() + "_" + board.name.title() +
-                            "_Init(uint32_t baudrate) {\n")
-                    f.write("  Can_Init(baudrate);\n")
+                    fw('void init_{}(uint32_t baudrate) '.format(coord(bus.name, board.name)))
+                    fw(
+                        '{' '\n'
+                        '\t' 'Can_Init(baudrate);' '\n'
```

Comment:
Small thing: I thought the policy (at least for MY18 code) was spaces for indentation. Why the change?

Reply:
CANlib is not MY18 code. :P It's always easy to find and replace '\t'. Maybe not so much for '    '. I picked tabs due to the nature of the project -- autogenerating code.